File: client/java/src/com/tibco/tgdb/pdu/impl/BeginTransactionRequest.java
/**
 * Copyright 2016 TIBCO Software Inc. All rights reserved.
 * 
 * Licensed under the Apache License, Version 2.0 (the "License"); You may not use this file except 
 * in compliance with the License.
 * A copy of the License is included in the distribution package with this file.
 * You also may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * <p/>
 * File name : BeginTransactionRequest.${EXT}
 * Created on: 2/4/15
 * Created by: suresh 
 * <p/>
 * SVN Id: $Id: BeginTransactionRequest.java 583 2016-03-15 02:02:39Z vchung $
 */


package com.tibco.tgdb.pdu.impl;

import com.tibco.tgdb.exception.TGException;
import com.tibco.tgdb.pdu.TGInputStream;
import com.tibco.tgdb.pdu.TGOutputStream;
import com.tibco.tgdb.pdu.VerbId;

import java.io.IOException;

public class BeginTransactionRequest extends AbstractProtocolMessage {

    @Override
    protected void writePayload(TGOutputStream os) throws TGException, IOException {

    }

    @Override
    protected void readPayload(TGInputStream is) throws TGException, IOException {

    }

    @Override
    public boolean isUpdateable() {
        return false;
    }

    @Override
    public VerbId getVerbId() {
        return VerbId.BeginTransactionRequest;
    }
}

File: client/nodejs/src/pdu/impl/BeginTransactionRequest.js
/**
 * Copyright 2016 TIBCO Software Inc. All rights reserved.
 * 
 * Licensed under the Apache License, Version 2.0 (the "License"); You may not use this file except 
 * in compliance with the License.
 * A copy of the License is included in the distribution package with this file.
 * You also may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

var util                    = require('util'),
    AuthenticatedMessage    = require('./AuthenticatedMessage').AuthenticatedMessage,
    AbstractProtocolMessage = require('../AbstractProtocolMessage').AbstractProtocolMessage,
    VerbId                  = require('./VerbId').VerbId;

//Class Definition
function BeginTransactionRequest() {
    AuthenticatedMessage.call();
}

util.inherits(BeginTransactionRequest, AuthenticatedMessage);

BeginTransactionRequest.prototype.getVerbId = function() {
    return VerbId.BEGIN_TRANS_REQUEST;
};

/**
 * Is this message updateable.
 */
BeginTransactionRequest.prototype.isUpdateable = function() {
    return false;
};

BeginTransactionRequest.prototype.writePayload = function(dynamicBuffer) {

};

BeginTransactionRequest.prototype.readPayload = function() {

};

BeginTransactionRequest.prototype.getClientId = function() {

};

BeginTransactionRequest.prototype.setClientId = function(clientId) {

};

exports.BeginTransactionRequest = BeginTransactionRequest;